core/vm/log.go

```diff
@@ -25,42 +25,47 @@ import (
 )
 
 type Log struct {
+	// Consensus fields
 	Address common.Address
 	Topics  []common.Hash
 	Data    []byte
-	Number  uint64
 
-	TxHash    common.Hash
-	TxIndex   uint
-	BlockHash common.Hash
-	Index     uint
+	// Derived fields (don't reorder!)
```

Comment:
Because `LogForStorage` is an automatic RLP encoder, which encodes the fields in order. So if you change the order, it will be incompatible with your current database contents. The reason I specifically added that comment is because the current field order is quite weird (block fields interrupted by transaction fields).